cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -111,15 +112,14 @@ public Integer call() throws Exception {
   }
 
   private Integer runningProgramCount(Id.Program program, String runId) throws Exception {
-    String path = String.format("apps/%s/workflows/%s/runs/%s/current", program.getApplicationId(), program.getId(),
-                                runId);
-    HttpResponse response = doGet(getVersionedAPIPath(path, program.getNamespaceId()));
-    if (response.getStatusLine().getStatusCode() == 200) {
-      String json = EntityUtils.toString(response.getEntity());
-      List<WorkflowActionNode> output = GSON.fromJson(json, LIST_WORKFLOWACTIONNODE_TYPE);
-      return output.size();
+    Map<String, WorkflowNodeStateDetail> nodeStates = getWorkflowNodeStates(program.toEntityId(), runId);
+    int count = 0;
+    for (WorkflowNodeStateDetail nodeStateDetail : nodeStates.values()) {
+      if (nodeStateDetail.getNodeStatus().equals(NodeStatus.RUNNING)) {
+        count++;
+      }
     }
-    return null;
+    return count;
```

Comment:
What happens to the case when response code is not 200

Reply:
in that case it will throw an exception, however in the test case, since we get the state for a program run with "run-id" and the "run-id" is obtained for running program. it wouldn't cause not found exception.